referenceframe/frame.go

```diff
@@ -630,3 +665,83 @@ func PoseToInputs(p spatial.Pose) []Input {
 		p.Orientation().OrientationVectorRadians().Theta,
 	})
 }
+
+// Determine whether two frames are (nearly) identical. For now, we only support implementers of
+// the frame interface that are registered (see register.go). However, this is not automatic, so
+// if a new implementer is registered manually in register.go, its case should be added here.
+//
+// NOTE: for ease, this function only takes one epsilon parameter because we have yet
+// to see a case of quantity where we want accept different levels of floating point error.
+// If the time comes where we want a different allowance for limits, vectors, and geometries,
+// this function should be changed accordingly.
+func framesAlmostEqual(frame1, frame2 Frame, epsilon float64) (bool, error) {
```

Comment:
In general I think it will be easier to be confident that all the cases I point out below are handled correctly if the concrete types that implement Frame (e.g. *staticFrame...)have an `(sf *staticFrame) Equal(other *staticFrame)` method which also handles the case when either sf and other are nil.